pkg/karmadactl/promote/promote.go

```diff
@@ -788,3 +745,66 @@ func buildClusterPropagationPolicy(resourceName, policyName, cluster string, gvr
 	}
 	return cpp
 }
+
+// ensureResourceInControlPlaneWithOptionalPolicy ensures that a specified resource is present in the Karmada control plane.
+// It optionally creates propagation policies for the resource.
+func ensureResourceInControlPlaneWithOptionalPolicy(controlPlaneDynamicClient dynamic.Interface, karmadaClient karmadaclientset.Interface,
```

Comment:
We are advised to use as few input parameters as possible, `isNamespaceScoped` can be calculated in the function.

```
func (o *CommandPromoteOption) ensureResourceInControlPlaneWithOptionalPolicy(controlPlaneDynamicClient dynamic.Interface, karmadaClient karmadaclientset.Interface,
	obj *unstructured.Unstructured, gvr schema.GroupVersionResource,
	propagateDeps bool)
```